src/swap/components/Swap.tsx

```diff
@@ -0,0 +1,110 @@
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { buildSwapTransaction } from '../core/buildSwapTransaction';
+import { getSwapQuote } from '../core/getSwapQuote';
+import { cn } from '../../lib/utils';
+import { SwapContext } from '../context';
+import type { SwapError, SwapReact } from '../types';
+import type { Token } from '../../token';
+
+function isSwapError(response: unknown): response is SwapError {
+  return response !== null && typeof response === 'object' && 'error' in response;
+}
+
+export function Swap({ account, children, onError, onSuccess }: SwapReact) {
+  const [fromAmount, setFromAmount] = useState('');
+  const [fromToken, setFromToken] = useState<Token>();
+  const [toAmount, setToAmount] = useState('');
+  const [toToken, setToToken] = useState<Token>();
+  const [swapQuoteError, setSwapQuoteError] = useState('');
+  const [swapTransactionError, setSwapTransactionError] = useState('');
+
+  const handleSubmit = useCallback(async () => {
+    setSwapQuoteError('');
+    setSwapTransactionError('');
+    if (account && fromToken && toToken && fromAmount) {
+      try {
+        const response = await buildSwapTransaction({
+          amount: fromAmount,
+          fromAddress: account.address,
+          from: fromToken,
+          to: toToken,
+        });
+        if (isSwapError(response)) {
+          setSwapTransactionError(response.error);
+          onError?.(response);
+        } else {
+          onSuccess?.(response);
+        }
+      } catch (error) {
+        console.log({ error });
+      }
+    }
+  }, [account, fromAmount, fromToken, toToken]);
+
+  const handleGetSwapQuote = useCallback(async () => {
+    setSwapQuoteError('');
+    setSwapTransactionError('');
+    if (fromToken && toToken && fromAmount) {
+      try {
+        const response = await getSwapQuote({ from: fromToken, to: toToken, amount: fromAmount });
+        if (isSwapError(response)) {
+          setSwapQuoteError(response.error);
+        } else {
+          setToAmount(response?.toAmount);
+        }
+      } catch (error) {
+        console.log({ error });
+      }
+    }
+  }, [fromAmount, fromToken, toToken]);
```

Comment:
This needs to work bidirectionally. Both edits to `fromAmount` and `toAmount` needs to trigger `handleGetSwapQuote`

perhaps we address in a follow up PR

Reply:
If it's not too much trouble, we should make the toAmount editable